### ISSUE TITLE
Errors with ETS5 file

### ISSUE THREAD

**Issue by MLat365** [laurent-martin/ets-to-homeassistant#14]
I'm new to Ruby and I've tried to use your script with an ETS5 file but it seems it cannot be opened:

c:\rubyknx>ruby ets_to_hass.rb --format homeass RueCourbe36-NouveauECS.knxproj > ha.yaml
<internal:C:/Program Files/Ruby32/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- zip (LoadError)
        from <internal:C:/Program Files/Ruby32/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from ets_to_hass.rb:6:in `<main>'

What's wrong? Maybe I don't do it rigth.

Thanks

**Comment by laurent-martin** [comment]
You need to install missings gems (ruby library)

**Comment by MLat365** [comment]
Thanh you.
Effectively I miss this step.
But has I just install Ruby, and never use it before, I need to find how to do this  and where this library need to be stored.

Cordialement,
M. Latour
________________________________

**Comment by laurent-martin** [comment]
No need to think about where to install, just execute in the main folder:

```bash
gem install bundler

bundle install
```

**Comment by MLat365** [comment]
I'm sure I do some mistakes as this doesn't work.

Here is all what I've done so far:

I'm running under Windows 7.
I've installed Ruby in C:\Ruby32
While installing msys32 it seems I have some errors as I did not have the Admin rigth but at the end it says it is installed.

I've created another directory C:\rubyknx where i have downloaded all the files from your [ets-to-homeassistant] repository
In the same directory I put my ETS Project file

I run "gem install bundler" from the cmd window
And it seems to work:
                    C:\rubyknx>gem install bundler
                    Fetching bundler-2.4.22.gem
                    Successfully installed bundler-2.4.22
                    Parsing documentation for bundler-2.4.22
                    Installing ri documentation for bundler-2.4.22
                    Done installing documentation for bundler after 1 seconds
                    1 gem installed

                   A new release of RubyGems is available: 3.4.10 → 3.4.22!
                   Run `gem update --system 3.4.22` to update your installation.

The I try to run "bundle install" but it doesn't work:
                  C:\rubyknx>bundle install
                  Could not locate Gemfile

Of course I'm sure that I'm doing wrong ;-)

**Comment by laurent-martin** [comment]
Well, the project contains a file named `Gemfile` , and the error says: no such file...
So, you did not download all files.
best, use the zip:

https://github.com/laurent-martin/ets-to-homeassistant/archive/refs/heads/master.zip

**Comment by MLat365** [comment]
Hi, 
So I look at the result.
I have created two Functions in the ETS Building structure in order to test.
In the output I found the two "Ligth" but:
- They are not associated with the Building Structure (Device/Entities)
- The Status GA is not taken into account
Here is the ETS and result file copy: 
![image](https://github.com/laurent-martin/ets-to-homeassistant/assets/152872711/a0cf5a5c-5fe7-4770-bdbc-a7356ff28a73)
Is it normal or is my ETS not correct?

**Comment by MLat365** [comment]
Hi,
Did you see my request?

**Comment by laurent-martin** [comment]
> Hi, Did you see my request?

Yes, for debugging, maybe you can send me the ETS project file, then I can run and see what's happening ?
(laurent.martin.l / gmail.com)

**Comment by MLat365** [comment]
Sure I can send the project file. Which direct address can I use?

**Comment by laurent-martin** [comment]
The address is up there in previous post, just replace / with @

**Comment by laurent-martin** [comment]
I have updated the README with additionnal information.
Especially, make sure that all group addresses that you want to use have a "data point type", i.e. group address type associated.
